edb/edgeql/compiler/schemactx.py

```diff
@@ -213,6 +213,9 @@ def derive_view(
                     stmtctx.pend_pointer_cardinality_inference(
                         ptrcls=ptr, ctx=ctx)
 
+    else:
+        raise RuntimeError("unsupported type in derive_view")
```

Comment:
`TypeError`?

It would be propagated as an `InternalServerError` anyways, but `TypeError` looks like a better choice here.

Reply:
I think we should just standardize on something like an `AssertionError` to stop wondering about which exception to use in case of an unexpected state on the insides.  I picked `AssertionError` because we also have lots of `assert` around.

Reply:
For cases like this I'd suggest something like `UnreachableError`.

Reply:
Or even `Never(BaseException)`

Reply:
`AssertionError` seems best to me if we want to standarize, since a lot of such errors are going to be assertion rerors